src/main/java/core/basesyntax/service/impl/FruitListImpl.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.FruitTransaction;
+import core.basesyntax.service.FruitList;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FruitListImpl implements FruitList {
+    @Override
+    public List<FruitTransaction> getFruitList(List<FruitTransaction> list) {
+        List<String> listOfFruitsString = list.stream()
+                .map(FruitTransaction::getFruit)
+                .distinct()
+                .toList();
+        List<FruitTransaction> listOfFruitsAsObjects = new ArrayList<>();
+        for (String fruit : listOfFruitsString) {
+            int defaultQuantity = 0;
+            listOfFruitsAsObjects.add(new FruitTransaction(fruit, defaultQuantity));
```

Comment:
The creation of 'FruitTransaction' with a default quantity of 0 may not be necessary if the 'FruitTransaction' class already provides a default constructor or a constructor that sets the quantity to 0. If that's the case, consider using that constructor instead.